controllers/bestie_controller.go

```diff
@@ -180,14 +180,14 @@ func (r *BestieReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	}
 
 	//Level 1 : update appVersion status.
-	appVersion := r.reportappversion(bestie)
+	appVersion := r.getDeployedBestieVersion(ctx, bestie)
 	if !reflect.DeepEqual(appVersion, bestie.Status.AppVersion) {
 		bestie.Status.AppVersion = appVersion
 		log.Info("update app version status")
 		err := r.Status().Update(ctx, bestie)
 		if err != nil {
 			log.Error(err, "Failed to update app-version status")
-			return ctrl.Result{Requeue: true}, err
+			return ctrl.Result{}, err
```

Comment:
I'm not sure but wondering if both statements result in requeue. We have many instances of ctrl.Result{Requeue: true}, err. Perhaps we should change them to ctrl.Result{}, err